app/models/king.rb

```diff
@@ -1,7 +1,51 @@
 class King < Piece
+  def valid_move?(to_row, to_col)
```

Comment:
I have to verify that the path between the king's current square and the rook's current square is not obstructed.

At one point, I did have the above move_obstructed? for a non-castle move. Then I realized that I had to call Piece::move_obstructed. I just couldn't figure out how to do that from the King class.